Tradução: data-visualize.qmd (#43)
cienciadedatos/pt-r4ds: data-visualize.qmd

Para facilitar a discussão, vamos definir alguns termos:

-   Uma **variável** é uma quantidade, qualidade ou propriedade que você pode medir.

-   Um **valor** é o estado de uma variável quando você a mede.
    O valor de uma variável pode mudar de medição para medição.

-   Uma **observação** é um conjunto de medições feitas em condições semelhantes (geralmente todas as medições em uma observação são feitas ao mesmo tempo e no mesmo objeto).
    Uma **observação** conterá vários valores, cada um associado a uma variável diferente.
    Às vezes, vamos nos referir a uma observação como um ponto de dados.

-   **Dados tabulares** são um conjunto de valores, cada um associado a uma variável e uma observação.
    Os dados tabulares estarão no formato ***tidy*** se cada valor estiver em sua própria "célula", cada variável em sua própria coluna e cada observação em sua própria linha.

Neste contexto, uma variável refere-se a um atributo de todos os pinguins, e uma observação refere-se a todos os atributos de um único pinguim.

Digite o nome do *data frame* no console e o R mostrará uma visualização de seu conteúdo.
Observe que aparece escrito `tibble` no topo desta visualização.

> No tidyverse, usamos *data frames* especiais chamados ***tibbles***, dos quais você aprenderá mais em breve.

Reviewer: nesse caso, me parece que os três asteriscos são para deixar em negrito e em itálico (e acho que faz sentido, pois no livro tá em negrito para destacar, mas não é uma palavra em português então deixamos em itálico.

#|   best fit line of the relationship between these two variables 
#|   overlaid. The plot displays a positive, fairly linear, and relatively 
#|   strong relationship between these two variables. especie (Adelie, 
#|   Chinstrap, and Gentoo) are represented with different colors and 
#|   shapes. The relationship between body mass and flipper length is 
#|   roughly the same for these three especie, and Gentoo pinguins are 
#|   larger than pinguins from the other two especie.

ggplot(pinguins, aes(x = comprimento_nadadeira, y = massa_corporal)) +
  geom_point(aes(color = especie, shape = especie)) +
  geom_smooth(method = "lm") +
  labs(
    title = "Massa corporal e comprimento da nadadeira",
    subtitle = "Medidas para Pinguim-de-adélia, Pinguim-de-barbicha e Pinguim-gentoo",
    x = "Comprimento da nadadeira (mm)",
    y = "Massa corporal (g)",
    color = "Espécie",
    shape = "Espécie"
  ) +
  scale_color_colorblind()
```

### Criando um ggplot

Vamos recriar esse gráfico passo a passo.

Com o ggplot2, você inicia um gráfico com a função `ggplot()`, definindo um objeto de gráfico ao qual você adiciona **camadas**.

> O primeiro argumento de `ggplot()` é o conjunto de dados a ser usado no gráfico e, portanto, `ggplot(data = pinguins)` cria um gráfico vazio que está preparado para exibir os dados dos `pinguins`, mas, como ainda não dissemos como visualizá-los, por enquanto ele está vazio.

Reviewer: ```suggestion
O primeiro argumento da função `ggplot()` é o conjunto de dados a ser usado no gráfico e, portanto, `ggplot(data = pinguins)` cria um gráfico vazio que está preparado para exibir os dados dos `pinguins`, mas, como ainda não dissemos como visualizá-los, por enquanto ele está vazio.
```

Esse não é um gráfico muito interessante, mas você pode pensar nele como uma tela vazia na qual você pintará as camadas restantes do seu gráfico.

```{r}
#| fig-alt: |
#|   Uma área de gráfico vazia em cinza

ggplot(data = pinguins)
```

Em seguida, precisamos informar ao `ggplot()` como as informações dos nossos dados serão representadas visualmente.
O argumento `mapping` (mapeamento) da função `ggplot()` define como as variáveis em seu conjunto de dados são mapeadas para as propriedades visuais (**estética**) do gráfico.

> O argumento `mapping` é sempre definido na função `aes()`, e os argumentos `x` e `y` de `aes()` especificam quais variáveis devem ser mapeadas para os eixos x e y.

Reviewer: ```suggestion
O argumento `mapping` é sempre definido na função `aes()`, e os argumentos `x` e `y` de `aes()` especificam quais variáveis devem ser mapeadas nos eixos x e y.
```

```{r}
#| warning: false
#| fig-alt: |
#|   Um gráfico de dispersão da massa corporal em função o comprimento da
#|   nadadeira dos pinguins. Sobreposta ao gráfico de dispersão está
#|   uma única linha de melhor ajuste que mostra a relação entre essas 
#|   variáveis para cada espécie (Pinguim-de-adélia, Pinguim-de-barbicha 
#|   e Pinguim-gentoo). As diferentes espécies de pinguins são plotadas 
#|   em cores diferentes somente para os pontos.

ggplot(
  data = pinguins,
  mapping = aes(x = comprimento_nadadeira, y = massa_corporal)
) +
  geom_point(mapping = aes(color = especie)) +
  geom_smooth(method = "lm")
```

Pronto!
Temos algo que se parece muito com nosso objetivo final, embora ainda não esteja perfeito.
Ainda precisamos usar formas diferentes para cada espécie de pinguim e melhorar os rótulos.

Geralmente, não é uma boa ideia representar informações usando apenas cores em um gráfico, pois as pessoas percebem as cores de forma diferente devido ao daltonismo ou a outras diferenças de visão de cores.
Portanto, além da cor, também podemos mapear `especie` para a estética `shape` (forma).

```{r}
#| warning: false
#| fig-alt: |

> #|   Um gráfico de dispersão da massa corporal em função o comprimento da

Reviewer: ```suggestion
#|   Um gráfico de dispersão da massa corporal em função do comprimento da
```

Um **boxplot** é um tipo de abreviação visual para medidas de posição (percentis) que descrevem uma distribuição.
Também é útil para identificar possíveis *outliers*.
Conforme mostrado em @fig-eda-boxplot, cada boxplot consiste em:

-   Uma caixa que indica o intervalo da metade intermediária dos dados, uma distância conhecida como intervalo interquartil (IIQ), que se estende do 25º percentil da distribuição até o 75º percentil.
    No meio da caixa há uma linha que exibe a mediana, ou seja, o 50º percentil, da distribuição.
    Essas três linhas lhe dão uma noção da dispersão da distribuição e se a distribuição é ou não simétrica em relação à mediana ou inclinada para um lado.

-   Pontos que apresentam observações com valores maiores que 1,5 vezes o IIQ de qualquer borda da caixa.
    Esses pontos discrepantes são incomuns e, por isso, são plotados individualmente.

-   Uma linha que se estende de cada extremidade da caixa e vai até o ponto mais distante (sem considerar os valores discrepantes - *outliers*) na distribuição.

```{r}
#| label: fig-eda-boxplot
#| echo: false
#| fig-cap: |

> #|   Diagrama sobre como um boxplot é criado.

Reviewer 2: ```suggestion
#|   Diagrama mostrando como um boxplot é criado.
```

#| fig-alt: |
#|   Um diagrama exibindo como um boxplot é criado seguindo os passos acima.

knitr::include_graphics("images/EDA-boxplot.png")
```

Vamos dar uma olhada na distribuição da massa corporal por espécie usando `geom_boxplot()`:

```{r}
#| warning: false
#| fig-alt: |
#|   Boxplots lado a lado da distribuição da massa corporal de 
#|   Pinguim-de-adélia, Pinguim-de-barbicha e Pinguim-gentoo. A 
#|   distribuição das massas corporais dos pinguins-de-Adélia e 
#|   Pinguim-de-barbicha parece ser simétrica, com medianas em torno de 
#|   3.750 gramas. A massa corporal mediana dos Pinguim-gentoo é muito 
#|   mais alta, em torno de 5.000 gramas, e a distribuição das massas 
#|   corporais desses pinguins parece ser um pouco inclinada para a 
#|   direita.

ggplot(pinguins, aes(x = especie, y = massa_corporal)) +
  geom_boxplot()
```

Como alternativa, podemos criar gráficos de densidade com `geom_density()`.

```{r}
#| warning: false
#| fig-alt: |
#|   Um gráfico de densidade da massa corporal de pinguins separado por 
#|   espécie. Cada espécie (Pinguim-de-adélia, Pinguim-de-barbicha e 

> #|   Pinguim-gentoo) é representada por um contorno colorido diferente em #|   cada curva de densidade.

Reviewer: ```suggestion
#|   Pinguim-gentoo) é representada por um contorno colorido diferente em
#|   cada curva de densidade.
```
aqui também é só uma quebra de linha

ggplot(pinguins, aes(x = massa_corporal, color = especie)) +
  geom_density(linewidth = 0.75)
```

Também personalizamos a espessura das linhas usando o argumento `linewidth` para que elas se destaquem um pouco mais contra o plano de fundo.

Além disso, podemos mapear `especie` para os atributos estéticos `color` e `fill` e usar o atributo `alpha` para adicionar transparência às curvas de densidade preenchidas.
Esse atributo assume valores entre 0 (completamente transparente) e 1 (completamente opaco).
No gráfico a seguir, ela está *definida* como 0.5.

```{r}
#| warning: false
#| fig-alt: |
#|   Um gráfico de densidade da massa corporal de pinguins separado por 
#|   espécie. Cada espécie (Pinguim-de-adélia, Pinguim-de-barbicha e 

> #|   Pinguim-gentoo) é representada por um contorno colorido diferente em #|   cada curva de densidade. As curvas de densidade também são 

Reviewer: ```suggestion
#|   Pinguim-gentoo) é representada por um contorno colorido diferente em 
#|   cada curva de densidade. As curvas de densidade também são 
```
quebrando linha

#|   to 6000.

ggplot(
  data = pinguins,
  mapping = aes(x = comprimento_nadadeira, y = massa_corporal)
)
```

Nossa tela vazia agora tem mais estrutura: está claro onde os comprimentos das nadadeiras serão exibidos (no eixo x) e onde as massas corporais serão exibidas (no eixo y).
Mas os pinguins em si ainda não estão no gráfico.
Isso ocorre porque ainda não definimos, em nosso código, como representar as observações de nosso *data frame* em nosso gráfico.

> Para isso, precisamos definir uma **geom**: A geometria que um gráfico usa para representar os dados.

Reviewer: ```suggestion
Para isso, precisamos definir um **geom**: a geometria que um gráfico usa para representar os dados.
```

Essas geometrias são disponibilizados no ggplot2 com funções que começam com `geom_`.
As pessoas geralmente descrevem os gráficos pelo tipo de geom que o gráfico usa.
Por exemplo, os gráficos de barras usam geometrias de barras (`geom_bar()`), os gráficos de linhas usam geometrias de linhas (`geom_line()`), os boxplots usam geometrias de boxplot (`geom_boxplot()`), os gráficos de dispersão usam geometrias de pontos (`geom_point()`) e assim por diante.

A função `geom_point()` adiciona uma camada de pontos ao seu gráfico, o que cria um gráfico de dispersão.
O ggplot2 vem com muitas funções de geometria, cada uma adicionando um tipo diferente de camada a um gráfico.
Você aprenderá várias geometrias ao longo do livro, principalmente em @sec-layers.

```{r}
#| fig-alt: |
#|   A scatterplot of body mass vs. flipper length of pinguins. The plot 
#|   displays a positive, linear, and relatively strong relationship between 
#|   these two variables.

ggplot(
  data = pinguins,
  mapping = aes(x = comprimento_nadadeira, y = massa_corporal)
) +
  geom_point()
```

Agora temos algo que se parece com o que poderíamos chamar de "gráfico de dispersão".

> Ele ainda não corresponde ao nosso gráfico de "objetivo final", mas, usando esse gráfico, podemos começar a responder à pergunta que motivou nossa exploração: "Como é a relação entre o comprimento da nadadeira e a massa corporal?" A relação parece ser positiva (à medida que o comprimento da nadadeira aumenta, a massa corporal também aumenta), razoavelmente linear (os pontos estão agrupados em torno de uma linha em vez de uma curva) e moderadamente forte (não há muita dispersão em torno dessa linha).

Reviewer: ```suggestion
Ele ainda não corresponde ao nosso gráfico mostrado no início da seção "objetivo final", mas, usando esse gráfico, podemos começar a responder à pergunta que motivou nossa exploração: "Como é a relação entre o comprimento da nadadeira e a massa corporal?" A relação parece ser positiva (à medida que o comprimento da nadadeira aumenta, a massa corporal também aumenta), razoavelmente linear (os pontos estão agrupados em torno de uma linha em vez de uma curva) e moderadamente forte (não há muita dispersão em torno dessa linha).
```

Os pinguins com nadadeiras mais longas geralmente são maiores em termos de massa corporal.

Antes de adicionarmos mais camadas a esse gráfico, vamos parar por um momento e revisar a mensagem de aviso que recebemos:

> Removed 2 rows containing missing values (`geom_point()`).

> Estamos vendo essa mensagem porque há dois pinguins em nosso conjunto de dados com valores ausentes de massa corporal e/ou comprimento da nadadeira e o ggplot2 não tem como representá-los no gráfico sem esses dois valores.

Reviewer: ```suggestion
Estamos vendo essa mensagem porque há dois pinguins em nosso conjunto de dados com valores ausentes (*NA*) de massa corporal e/ou comprimento da nadadeira e o ggplot2 não tem como representá-los no gráfico sem esses dois valores.
```

> Assim como o R, o ggplot2 adota a filosofia de que os valores ausentes nunca devem desaparecer silenciosamente.

Reviewer: ```suggestion
Assim como o R, o ggplot2 adota a filosofia de que os valores faltantes nunca devem desaparecer silenciosamente.
```

> Esse tipo de aviso é provavelmente um dos tipos mais comuns de avisos que você verá ao trabalhar com dados reais - os valores ausentes são um problema muito comum e você aprenderá mais sobre eles ao longo do livro, especialmente em @sec-missing-values.

Reviewer: ```suggestion
Esse tipo de aviso é provavelmente um dos tipos mais comuns de avisos que você verá ao trabalhar com dados reais - os valores faltantes são um problema muito comum e você aprenderá mais sobre eles ao longo do livro, especialmente em @sec-missing-values.
```

> Nos demais gráficos deste capítulo, suprimiuprimir esse aviso para que ele não seja mostrado em cada gráfico que fizermos.

Reviewer: ```suggestion
Nos demais gráficos deste capítulo, vamos suprimir esse aviso para que ele não seja mostrado em cada gráfico que fizermos.
```

### Adicionando atributos estéticos e camadas {#sec-adding-aesthetics-layers}

> Gráficos de dispersão são úteis para exibir a relação entre duas variáveis numéricas, mas é sempre uma boa ideia ser cético em relação a qualquer relação aparente entre duas variáveis e perguntar se pode haver outras variáveis que expliquem ou mudem a natureza dessa relação aparente.

Reviewer: ```suggestion
Gráficos de dispersão são úteis para exibir a relação entre duas variáveis numéricas, mas é sempre uma boa ideia ter uma postura cético em relação a qualquer relação aparente entre duas variáveis e perguntar se pode haver outras variáveis que expliquem ou mudem a natureza dessa relação aparente.
```

```{r}
#| eval: false

ggplot(
  data = pinguins,
  mapping = aes(x = comprimento_nadadeira, y = massa_corporal, color = ilha)
) +
  geom_point() +
  geom_smooth(se = FALSE)
```

10. Esses dois gráficos serão diferentes?
    Por que sim ou por que não?

    ```{r}
    #| eval: false

    ggplot(
      datai = pinguins,
      mapping = aes(x = comprimento_nadadeira, y = massa_corporal)
    ) +
      geom_point() +
      geom_smooth()

    ggplot() +
      geom_point(
        data = pinguins,
        mapping = aes(x = comprimento_nadadeira, y = massa_corporal)
      ) +
      geom_smooth(
        data = pinguins,
        mapping = aes(x = comprimento_nadadeira, y = massa_corporal)
      )
    ```

## Chamadas ggplot2 {#sec-ggplot2-calls}

À medida que passarmos dessas seções introdutórias, faremos a transição para uma expressão mais concisa do código do ggplot2.
Até agora, temos sido muito explícitos, o que é útil quando se está aprendendo:

```{r}
#| eval: false

ggplot(
  data = pinguins,
  mapping = aes(x = comprimento_nadadeira, y = massa_corporal)
) +
  geom_point()
```

> Normalmente, o primeiro ou os dois primeiros argumentos de uma função são tão importantes que você deve sabê-los de cor.

Reviewer 2: ```suggestion
Normalmente, o primeiro ou os dois primeiros argumentos de uma função são tão importantes que você deve saber usar de cor.
```

> Normalmente, o primeiro ou os dois primeiros argumentos de uma função são tão importantes que você deve sabê-los de cor.

Reviewer: ```suggestion
Normalmente, o primeiro ou os dois primeiros argumentos de uma função são tão importantes que você logo saberá usar eles de cor.
```

Reviewer: Essa minha sugestão muda um pouco do sentido da frase, e é uma questão pessoal minha... sinto que o "deve" fica muito pesado!

> Os dois primeiros argumentos de `ggplot()` são `data` e `mapping`; no restante do livro, não forneceremos esses nomes.

Reviewer: ```suggestion
Os dois primeiros argumentos de `ggplot()` são `data` e `mapping`; no restante do livro, não escreveremos esses nomes.
```

Isso economiza digitação e, ao reduzir a quantidade de texto extra, facilita a visualização das diferenças entre os gráficos.
Essa é uma preocupação de programação realmente importante, à qual voltaremos em @sec-functions.

Reescrevendo o gráfico anterior de forma mais concisa, temos:

```{r}
#| eval: false

ggplot(pinguins, aes(x = comprimento_nadadeira, y = massa_corporal)) +
  geom_point()
```

> No futuro, você também aprenderá sobre o pipe, `|>`, que permitirá que você crie esse gráfico com:

Reviewer 2: ```suggestion
No futuro, você também aprenderá sobre o pipe(encadeamento), `|>`, que permitirá que você crie esse gráfico com a seguinte sintaxe:
```

#|   and right skewed, ranging between approximately 2500 to 6500 grams.

ggplot(pinguins, aes(x = massa_corporal)) +
  geom_histogram(binwidth = 200)
```

Um histograma divide o eixo x em intervalos igualmente espaçados e, em seguida, usa a altura de uma barra para exibir o número de observações que se enquadram em cada intervalo.
No gráfico acima, a barra mais alta mostra que 39 observações têm um valor `massa_corporal` entre 3500 e 3700 gramas, que são as bordas esquerda e direita da barra.

Você pode definir a largura dos intervalos em um histograma com o argumento *binwidth* (largura do intervalo), que é medido nas unidades da variável `x`.
Você deve sempre explorar uma variedade de larguras de intervalos ao trabalhar com histogramas, pois diferentes larguras de intervalos podem revelar padrões diferentes.
Nos gráficos abaixo, uma largura de intervalo de 20 é muito estreita, resultando em muitas barras, o que dificulta a determinação da forma da distribuição.
Da mesma forma, uma largura de intervalo de 2000 é muito alta, resultando em todos os dados sendo agrupados em apenas três barras, o que também dificulta a determinação da forma da distribuição.

> Uma largura de intervalo de 200 proporciona um balanço adequado.

Reviewer 2: ```suggestion
Uma largura de intervalo de 200 proporciona um balanço mais adequado.
```

```{r}
#| warning: false
#| layout-ncol: 2
#| fig-width: 3
#| fig-alt: |
#|   Two histograms of body masses of pinguins, one with binwidth of 20 
#|   (left) and one with binwidth of 2000 (right). The histogram with binwidth 
#|   of 20 shows lots of ups and downs in the heights of the bins, creating a 
#|   jagged outline. The histogram  with binwidth of 2000 shows only three bins.

ggplot(pinguins, aes(x = massa_corporal)) +
  geom_histogram(binwidth = 20)
ggplot(pinguins, aes(x = massa_corporal)) +
  geom_histogram(binwidth = 2000)
```

Uma visualização alternativa para distribuições de variáveis numéricas é um gráfico de densidade.

> Um gráfico de densidade é uma versão suavizada de um histograma e uma alternativa prática, especialmente para dados contínuos provenientes de uma distribuição suavizada subjacente.

Reviewer: eu achei meio confuso o que quer dizer com o "distribuição suavizada subjacente"

Não entraremos em detalhes sobre como `geom_density()` estima a densidade (você pode ler mais sobre isso na documentação da função), mas vamos explicar como a curva de densidade é desenhada com uma analogia.
Imagine um histograma feito de blocos de madeira.
Em seguida, imagine que você jogue um fio de espaguete cozido sobre ele.
A forma que o espaguete assumirá sobre os blocos pode ser considerada como a forma da curva de densidade.

> Ela mostra menos detalhes do que um histograma, mas pode facilitar a obtenção rápida da forma da distribuição, principalmente com relação aos modos e à assimetria.

Reviewer: Acho que modes aqui está falando dos números mais frequentes, não?
```suggestion
Ela mostra menos detalhes do que um histograma, mas pode facilitar a obtenção rápida da forma da distribuição, principalmente com relação à moda (valor que ocorre com maior frequência) e à assimetria.
```

```{r}
#| fig-alt: |
#|   A density plot of body masses of pinguins. The distribution is unimodal 
#|   and right skewed, ranging between approximately 2500 to 6500 grams.

ggplot(pinguins, aes(x = massa_corporal)) +
  geom_density()
```

### Exercícios

1.  Faça um gráfico de barras de `especie` de `pinguins`, no qual você atribui `especie` ao atributo estético `y`.
    Como esse gráfico é diferente?

2.  Como os dois gráficos a seguir são diferentes?
    Qual atributo estético, `color` ou `fill`, é mais útil para alterar a cor das barras?

```{r}
#| eval: false

ggplot(pinguins, aes(x = especie)) +
  geom_bar(color = "red")

ggplot(pinguins, aes(x = especie)) +
  geom_bar(fill = "red")
```

3.  O que o argumento `bins` em `geom_histogram()` faz?

4.  Faça um histograma da variável `quilate` no conjunto de dados `diamante` que está disponível quando você carrega o pacote dados.
    Faça experiências com diferentes larguras de intervalo (*binwidth*).
    Qual largura de intervalo revela os padrões mais interessantes?

## Visualizando relações

Para visualizar uma relação, precisamos ter pelo menos duas variáveis mapeadas para os atributos estéticos de um gráfico.

> Nas seções a seguir, você aprenderá sobre os gráficos comumente usados para visualizar relações entre duas ou mais variáveis e os geoms usados para criá-los.

Reviewer: ```suggestion
Nas seções a seguir, você aprenderá sobre os gráficos comumente usados para visualizar relações entre duas ou mais variáveis e as geometrias usados para criá-los.
```

### Uma variável numérica e uma variável categórica

Para visualizar a relação entre uma variável numérica e uma variável categórica, podemos usar diagramas de caixa (chamados ***boxplots***) lado a lado.
Um **boxplot** é um tipo de abreviação visual para medidas de posição (percentis) que descrevem uma distribuição.
Também é útil para identificar possíveis *outliers*.
Conforme mostrado em @fig-eda-boxplot, cada boxplot consiste em:

-   Uma caixa que indica o intervalo da metade intermediária dos dados, uma distância conhecida como intervalo interquartil (IIQ), que se estende do 25º percentil da distribuição até o 75º percentil.
    No meio da caixa há uma linha que exibe a mediana, ou seja, o 50º percentil, da distribuição.
    Essas três linhas lhe dão uma noção da dispersão da distribuição e se a distribuição é ou não simétrica em relação à mediana ou inclinada para um lado.

-   Pontos que apresentam observações com valores maiores que 1,5 vezes o IIQ de qualquer borda da caixa.
    Esses pontos discrepantes são incomuns e, por isso, são plotados individualmente.

> -   Uma linha que se estende de cada extremidade da caixa e vai até o ponto mais distante sem *outlier* na distribuição.

Reviewer: ```suggestion
-   Uma linha que se estende de cada extremidade da caixa e vai até o ponto mais distante (sem considerar os valores discrepantes - *outliers*) na distribuição.
```

#|   Chinstrap pinguins' body masses appear to be symmetric with 
#|   medians around 3750 grams. The median body mass of Gentoo pinguins 
#|   is much higher, around 5000 grams, and the distribution of the 
#|   body masses of these pinguins appears to be somewhat right skewed.

ggplot(pinguins, aes(x = especie, y = massa_corporal)) +
  geom_boxplot()
```

Como alternativa, podemos criar gráficos de densidade com `geom_density()`.

```{r}
#| warning: false
#| fig-alt: |
#|   A density plot of body masses of pinguins by especie of pinguins. Each 
#|   especie (Adelie, Chinstrap, and Gentoo) is represented with different 
#|   colored outlines for the density curves.

ggplot(pinguins, aes(x = massa_corporal, color = especie)) +
  geom_density(linewidth = 0.75)
```

Também personalizamos a espessura das linhas usando o argumento `linewidth` para que elas se destaquem um pouco mais contra o plano de fundo.

Além disso, podemos mapear `especie` para os atributos estéticos `color` e `fill` e usar o atributo `alpha` para adicionar transparência às curvas de densidade preenchidas.
Esse atributo assume valores entre 0 (completamente transparente) e 1 (completamente opaco).

> No gráfico a seguir, ela está *definida* como 0,5.

Reviewer: ```suggestion
No gráfico a seguir, ela está *definida* como 0.5.
```

```{r}
#| warning: false
#| fig-alt: |
#|   A density plot of body masses of pinguins by especie of pinguins. Each 
#|   especie (Adelie, Chinstrap, and Gentoo) is represented in different 
#|   colored outlines for the density curves. The density curves are also 
#|   filled with the same colors, with some transparency added.

ggplot(pinguins, aes(x = massa_corporal, color = especie, fill = especie)) +
  geom_density(alpha = 0.5)
```

Observe a terminologia que usamos aqui:

> -   Nós *mapeamos* variáveis para atributos estéticos se quisermos que o atributo visual representado por esse atributo varie com base nos valores dessa variável.

Reviewer: ```suggestion
-   Nós *mapeamos* variáveis para atributos estéticos se quisermos que o atributo visual representado por esse atributo varie de acordo com os valores dessa variável.
```

> -   Caso contrário, nós *definimos* o valor de um atributo estético.

Reviewer: ```suggestion
-   Caso contrário, nós **definimos** o valor de um atributo estético.
```

### Duas variáveis categóricas

Podemos usar gráficos de barras empilhadas para visualizar a relação entre duas variáveis categóricas.
Por exemplo, os dois gráficos de barras empilhadas a seguir exibem a relação entre `ilha` e `espécie` ou, especificamente, a visualização da distribuição de `espécie` em cada ilha.

O primeiro gráfico mostra as frequências de cada espécie de pinguim em cada ilha.
O gráfico de frequências mostra que há um número igual de Pinguim-de-adélia em cada ilha.
Mas não temos uma boa noção do equilíbrio percentual em cada ilha.

```{r}
#| fig-alt: |
#|   Bar plots of penguin especie by ilha (Biscoe, Dream, and Torgersen)
ggplot(pinguins, aes(x = ilha, fill = especie)) +
  geom_bar()
```

> O segundo gráfico é um gráfico de frequência relativa, criado pela definição de `position = "fill"` na geom, que é mais útil para comparar as distribuições de espécies entre as ilhas, pois não é afetado pelo número desigual de pinguins entre as ilhas.

Reviewer: ```suggestion
O segundo gráfico é um gráfico de frequência relativa, criado pela definição de `position = "fill"` na geometria, que é mais útil para comparar as distribuições de espécies entre as ilhas, pois não é afetado pelo número desigual de pinguins entre as ilhas.
```
"na geom" pra mim fica estranho mas tbm não sei como é a melhor forma..

> Usando esse gráfico, podemos ver que todos os Pinguim-gentoo vivem na ilha Biscoe e constituem aproximadamente 75% dos pinguins dessa ilha, todos os Pinguim-de-barbicha vivem na ilha Dream e constituem aproximadamente 50% dos pinguins dessa ilha, e os Pinguim-de-adélia vivem nas três ilhas e constituem todos os pinguins de Torgersen.

Reviewer: ```suggestion
Usando esse gráfico, podemos ver que todos os Pinguim-gentoo vivem na ilha Biscoe e constituem aproximadamente 75% dos pinguins dessa ilha, todos os Pinguim-de-barbicha vivem na ilha Dream e constituem aproximadamente 50% dos pinguins dessa ilha, e os Pinguim-de-adélia vivem nas três ilhas e constituem todos os pinguins da ilha Torgersen.
```

#|   displays a positive, linear, relatively strong relationship between 
#|   these two variables.

ggplot(pinguins, aes(x = comprimento_nadadeira, y = massa_corporal)) +
  geom_point()
```

### Três ou mais variáveis

Como vimos em @sec-adding-aesthetics-layers, podemos incorporar mais variáveis em um gráfico mapeando-as para atributos estéticos adicionais.
Por exemplo, no gráfico de dispersão a seguir, as cores dos pontos (*color*) representam espécies e as formas dos pontos (*shape*) representam ilhas.

```{r}
#| warning: false
#| fig-alt: |
#|   A scatterplot of body mass vs. flipper length of pinguins. The plot 
#|   displays a positive, linear, relatively strong relationship between 
#|   these two variables. The points are colored based on the especie of the 
#|   pinguins and the shapes of the points represent ilhas (round points are 
#|   Biscoe ilha, triangles are Dream ilha, and squared are Torgersen 
#|   ilha). The plot is very busy and it's difficult to distinguish the shapes
#|   of the points.

ggplot(pinguins, aes(x = comprimento_nadadeira, y = massa_corporal)) +
  geom_point(aes(color = especie, shape = ilha))
```

> No entanto, mapear muitos atirbutos estéticos a um gráfico faz com que ele fique desordenado e difícil de entender.

Reviewer 2: ```suggestion
No entanto, mapear muitos atributos estéticos a um gráfico faz com que ele fique desordenado e difícil de entender.
```

> Outra maneira, que é particularmente útil para variáveis categóricas, é dividir seu gráfico em **facetas**, subdivisões que exibem um subconjunto dos dados cada uma.

Reviewer 2: ```suggestion
Outra maneira, que é particularmente útil para variáveis categóricas, é dividir seu gráfico em **facetas**, subdivisões ou janelas que exibem um subconjunto dos dados cada uma.
```

> Outra maneira, que é particularmente útil para variáveis categóricas, é dividir seu gráfico em **facetas**, subdivisões que exibem um subconjunto dos dados cada uma.

Reviewer: ```suggestion
Outra maneira, que é particularmente útil para variáveis categóricas, é dividir seu gráfico em **facetas** (*facets*), subdivisões que exibem um subconjunto dos dados cada uma.
```

5.  Faça um gráfico de dispersão de `profundidade_bico` vs. `comprimento_bico` e pinte os pontos por `especie`.
    O que a adição da coloração por especie revela sobre a relação entre essas duas variáveis?
    E quanto à separação em facetas por `especie`?

6.  Por que o seguinte código produz duas legendas separadas?
    Como você corrigiria isso para combinar as duas legendas?

```{r}
#| warning: false
#| fig-show: hide

ggplot(
  data = pinguins,
  mapping = aes(
    x = comprimento_bico, y = profundidade_bico, 
    color = especie, shape = especie
  )
) +
  geom_point() +
  labs(color = "especie")
```

7.  Crie os dois gráficos de barras empilhadas a seguir. Que pergunta você pode responder com o primeiro? Que pergunta você pode responder com o segundo?

```{r}
#| fig-show: hide

ggplot(pinguins, aes(x = ilha, fill = especie)) +
  geom_bar(position = "fill")
ggplot(pinguins, aes(x = especie, fill = ilha)) +
  geom_bar(position = "fill")
```

## Salvando seus gráficos {#sec-ggsave}

Depois de criar um gráfico, talvez você queira tirá-lo do R salvando-o como uma imagem que possa ser usada em outro lugar.

> Essa é a função de `ggsave()`, que salvará no computador o gráfico criado mais recentemente:

Reviewer: ```suggestion
Esse é o objetido da função `ggsave()`, que salvará no computador o gráfico criado mais recentemente:
```

file.remove("penguin-plot.png")
```

Isso salvará o gráfico no seu diretório de trabalho, um conceito sobre o qual você aprenderá mais em @sec-workflow-scripts-projects.

Se você não especificar a largura `width` e a altura `height`, elas serão tiradas das dimensões do dispositivo de plotagem atual.
Para obter um código reprodutível, você deverá especificá-los.

> Você pode obter mais informações sobre `ggsave()` na documentação.

Reviewer: ```suggestion
Você pode obter mais informações sobre a função `ggsave()` na documentação.
```

> De modo geral, entretanto, recomendamos que você monte seus relatórios finais usando o Quarto, um sistema de autoria reproduzível que permite intercalar seu código e sua escrita e incluir automaticamente seus gráficos em seus relatórios.

Reviewer 2: ```suggestion
De modo geral, entretanto, recomendamos que você monte seus relatórios finais usando o Quarto, um sistema de editoração técnico científica que permite intercalar seu código e sua escrita e incluir automaticamente seus gráficos em seus relatórios.
```

Reviewer 2: Achei essa parte do Quarto difícil de localizar para o português, acho que a @beatrizmilz pode contribuir aqui. De qualquer forma deixei uma proposta para o termo em pt-br.

Reviewer: Acho que em vez de "autoria", poderia ser "escrita". 
sobre o termo original usado (reproducible), acho que o principal é a ideia de poder "gerar novamente" o arquivo e ter os mesmos resultados..
por mais que o quarto seja uma ferramenta de escrita técnica e científica, nem todas as ferramentas de escrita técnica e científica geram arquivos reprodutíveis, certo?

Eu acho que a sugestão do @nextmarte fica mais fácil de entender, mas tbm sai um pouco do sentido do termo original dos autores (reproducible).

em resumo: não sei o que é melhor HAHHAHA

@clente qual é a sua opinião, grande mestre?

Reviewer 2: "Um sistema de escrita técnico-científica que permite reprodutibilidade, ou seja, permite que outras pessoas consigam reproduzir o que foi escrito" 

O que vocês acham?

Reviewer 3: Neste contexto, "técnico-científica" foge do significado original; podemos usar o Quarto para qualquer coisa, mesmo escrita que não seja técnico-científica.

Eu pessoalmente traduziria como "escrita reprodutível", mas também sou favorável ao "autoria reproduzível" original (ambos "reprodutível" e "reproduzível" são válidos). A @rmhirota sugeriu "criação" ao invés de "escrita" ou "autoria".

Author: Concordo que "técnico-científica" foge do significado pois muitos utilizam o Quarto para fazer sites, tutoriais, etc.
Deixei "escrita reprodutível" pois engloba todos estes significados.

> Você saberá mais sobre o Quarto em @sec-quarto.

Reviewer: ```suggestion
Você aprenderá mais sobre o Quarto em @sec-quarto.
```

Ao começar a executar o código em R, é provável que você encontre problemas.
Não se preocupe, isso acontece com todo mundo.
Todos nós estamos escrevendo código em R há anos, mas todos os dias ainda escrevemos códigos que não funciona na primeira tentativa!

Comece comparando cuidadosamente o código que está executando com o código do livro.
O R é extremamente exigente, e um caractere mal colocado pode fazer toda a diferença.
Certifique-se de que cada `(` seja combinado com um `)` e que cada `"` seja combinado com outra `"`.
Às vezes, você executará o código e nada acontecerá.
Verifique o lado esquerdo do console: se houver um `+`, isso significa que o R acha que você não digitou uma expressão completa e está esperando que você a termine.
Nesse caso, geralmente é fácil começar do zero novamente pressionando *Esc* para interromper o processamento do comando atual.

Um problema comum ao criar gráficos ggplot2 é colocar o `+` no lugar errado: ele deve vir no final da linha, não no início.
Em outras palavras, certifique-se de não ter escrito acidentalmente um código como este:

```{r}
#| eval: false

ggplot(data = milhas) 
+ geom_point(mapping = aes(x = cilindrada, y = rodovia))
```

> Se você ainda estiver com dificuldades, tente a ajuda.

Reviewer: ```suggestion
Se você ainda estiver com dificuldades, tente a ajuda (painel *Help*).
```

Você pode obter ajuda sobre qualquer função do R executando `?nome_da_função` no console ou selecionando o nome da função e pressionando F1 no RStudio.
Não se preocupe se a ajuda não parecer muito útil - em vez disso, pule para os exemplos e procure um código que corresponda ao que você está tentando fazer.

Se isso não ajudar, leia atentamente a mensagem de erro.
Às vezes, a resposta estará escondida lá!

> Mas quando você é novo no R, mesmo que a resposta esteja na mensagem de erro, talvez você ainda não saiba como entendê-la.

Reviewer: ```suggestion
Mas quando você é iniciante no R, mesmo que a resposta esteja na mensagem de erro, talvez você ainda não saiba como entendê-la.
```

Outra ferramenta excelente é o Google: tente pesquisar a mensagem de erro no Google, pois é provável que outra pessoa tenha tido o mesmo problema e tenha obtido ajuda on-line.

## Resumo

Neste capítulo, você aprendeu os fundamentos da visualização de dados com o ggplot2.
Começamos com a ideia básica que sustenta o ggplot2: uma visualização é um mapeamento de variáveis em seus dados para atributos estéticos como posição (*position*), cor (*color*), tamanho (*size*) e forma (*shape*).
Em seguida, você aprendeu a aumentar a complexidade e melhorar a apresentação de seus gráficos camada por camada.

> Você também aprendeu sobre gráficos comumente usados para visualizar a distribuição de uma única variável, bem como para visualizar relações entre duas ou mais variáveis ao utilizar mapeamentos de atributos estéticos adicionais e/ou dividindo seu gráfico em pequenos múltiplos usando facetas.

Reviewer: ```suggestion
Você também aprendeu sobre gráficos comumente usados para visualizar a distribuição de uma única variável, bem como para visualizar relações entre duas ou mais variáveis ao utilizar mapeamentos de atributos estéticos adicionais e/ou dividindo seu gráfico em pequenos gráficos usando facetas (*facets*).
```

Reviewer: não sei se ficou boa a sugestão. mas tá confuso pra mim a forma original

Usaremos as visualizações repetidamente ao longo deste livro, introduzindo novas técnicas à medida que precisarmos delas, além de nos aprofundarmos na criação de visualizações com o ggplot2 em @sec-layers por meio da @sec-communication.

Com as noções básicas de visualização em seu currículo, no próximo capítulo mudaremos um pouco a direção e daremos algumas orientações práticas sobre o fluxo de trabalho.

> Intercalamos conselhos sobre fluxo de trabalho com ferramentas de ciência de dados ao longo desta parte do livro, pois isso o ajudará a se manter organizado à medida que você escreve quantidades cada vez maiores de código em R.

Reviewer: ```suggestion
Intercalamos conselhos sobre fluxo de trabalho com ferramentas de ciência de dados ao longo desta parte do livro, pois isso te ajudará a se manter organizado à medida que você escreve quantidades cada vez maiores de código em R.
```
